### PR TITLE
Upgrade/800/server profile template

### PR DESCRIPTION
**Pull request by sijeesh**
### Description
Upgraded Server Profile Template resource to support OneView API version 800

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass for Python 2.7+ & 3.4+(`$ tox`).
- [ ] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [x] New endpoints supported are updated in the endpoints-support.md file.
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: examples/server_profile_templates.py [hunk 1]
```diff
@@ -26,105 +26,106 @@
...
 
+# Get Server Profile Template by scope_uris
+if oneview_client.api_version >= 600:
+    server_profile_templates_by_scope_uris = profile_templates.get_all(
+        scope_uris="\"'/rest/scopes/3bb0c754-fd38-45af-be8a-4d4419de06e9'\"")
+    if len(server_profile_templates_by_scope_uris) > 0:
+        print("Found %d Server profile Templates" % (len(server_profile_templates_by_scope_uris)))
+        i = 0
+        while i < len(server_profile_templates_by_scope_uris):
+            print("Found Server Profile Template Group by scope_uris: '%s'.\n  uri = '%s'" % (server_profile_templates_by_scope_uris[i]['name'],
```
Comment (soodpr) [inline examples/server_profile_templates.py]: remove group keyword from the print

File: examples/server_profile_templates.py [hunk 2]
```diff
@@ -26,105 +26,106 @@
...
+                                                                                              server_profile_templates_by_scope_uris[i]['uri']))
+            i += 1
+        pprint(server_profile_templates_by_scope_uris)
+    else:
+        print("No Server Profile Template Group found.")
```
Comment (soodpr) [inline examples/server_profile_templates.py]: same as above

File: tests/unit/resources/servers/test_server_profile_templates.py [hunk 3]
```diff
@@ -49,89 +51,53 @@ def test_get_all(self, mock_get_all):
...
-    @mock.patch.object(ResourceClient, 'delete')
-    def test_delete(self, mock_delete):
-        template = dict(name="BL460c Gen8 1")
+        mock_update.assert_called_once_with(template, self.uri, True, -1)
 
-        self._resource.delete(resource=template, timeout=TIMEOUT)
-        mock_delete.assert_called_once_with(resource=template, timeout=TIMEOUT, force=False)
+    @mock.patch.object(ResourceHelper, 'delete')
+    def test_delete(self, mock_delete):
+        self._resource.delete(timeout=TIMEOUT)
```
Comment (dileepds) [inline tests/unit/resources/servers/test_server_profile_templates.py]: Just a doubt, why 'template' is not passing for delete call where as it is passing for create ? I saw similar calls for 'delete' in other PRs also.

Comment (sijeesh) [reply tests/unit/resources/servers/test_server_profile_templates.py]: Create method needs data to create the resource and it returns the object of the created resource.
to delete a resource we have to access that resource object using name/URI and delete it. Eg obj.delete(query param).

File: examples/server_profile_templates.py (hunk 1 above)
Comment (sijeesh) [reply examples/server_profile_templates.py]: Removed

File: examples/server_profile_templates.py (hunk 2 above)
Comment (sijeesh) [reply examples/server_profile_templates.py]: Removed

**Pull request merged**